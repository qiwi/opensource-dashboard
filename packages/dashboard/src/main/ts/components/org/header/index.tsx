import { Box, Flex, FlexItem, Lnk } from '@qiwi/pijma-core'
import { Header as PijmaHeader, HeaderMenu } from '@qiwi/pijma-desktop'
import React from 'react'
import { useHistory, useLocation } from 'react-router-dom'

export const Header = () => {
  const { pathname } = useLocation()
  const history = useHistory()
  return (
    <PijmaHeader underline={true}>
      <Flex width={1} height={1} justify="space-between">
        <FlexItem ml={6}>
          <Flex height={1}>
            <FlexItem align="center" shrink={0} mr={11}>
              <Lnk href={window.location.href}>
                <Box
                  as="img"
                  // @ts-ignore
                  src="https://static.qiwi.com/img/qiwi_com/header/qiwi-wallet-logo.svg"
                  width={24}
                  height={12}
                />
              </Lnk>
            </FlexItem>
            <FlexItem shrink={0} mr={6}>
              <HeaderMenu>
                {[
                  {
                    title: 'Reporter',
                    active: pathname === '/',
                    onClick: () => history.push('/'),
                  },
                  {
                    title: 'Crawler',
                    onClick: () => history.push('/crawler'),
                    active: pathname === '/crawler',
                  },
                ]}
              </HeaderMenu>
            </FlexItem>
          </Flex>
        </FlexItem>
      </Flex>
    </PijmaHeader>
  )
}
