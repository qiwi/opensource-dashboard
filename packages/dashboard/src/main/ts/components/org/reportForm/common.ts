export type TReportProps = {
  data: {
    type: string
    depType: string
    packageNamePattern: string
    source: string
    versionRange?: string
  }
  setData: (args: Record<string, any>) => void
}
