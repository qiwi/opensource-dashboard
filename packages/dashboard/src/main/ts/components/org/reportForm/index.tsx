export { TreeReportForm } from './treeReportForm'
export { UsageReportForm } from './usageReportForm'
export { VersionsReportForm } from './versionsReportForm'
export type { TReportProps } from './common'
