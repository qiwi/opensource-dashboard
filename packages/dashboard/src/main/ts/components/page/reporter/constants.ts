export const treeDefault = {
  type: 'tree',
  depType: 'default',
  packageNamePattern: '',
  source: 'auto',
}

export const versionsDefault = {
  type: 'versions',
  depType: 'default',
  packageNamePattern: '',
  source: 'auto',
  versionRange: '',
}

export const usageDefault = {
  type: 'usage',
  depType: 'default',
  packageNamePattern: '',
  source: 'auto',
  versionRange: '',
}

export const defaultTypes = {
  tree: treeDefault,
  usage: usageDefault,
  versions: versionsDefault,
}
