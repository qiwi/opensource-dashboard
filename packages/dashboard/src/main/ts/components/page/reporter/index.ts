export {
  versionsDefault,
  treeDefault,
  usageDefault,
  defaultTypes,
} from './constants'
export { Reporter } from './reporter'
