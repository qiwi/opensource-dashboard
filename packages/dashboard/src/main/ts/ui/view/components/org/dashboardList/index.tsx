import { Card, FlexItem, Striper } from '@qiwi/pijma-core'
import { BlockContent } from '@qiwi/pijma-desktop'
import React, { useEffect, useState } from 'react'

import { treeReportService } from '../../../../../service'

export const DashboardList = () => {
  const [report, setReport] = useState([])
  useEffect(() => {
    // @ts-ignore
    treeReportService.getTreeReport().then(data => setReport(data))
  }, [])

  return (
    <Card m={20} r={12} pl={12} pr={12} pt={4} pb={4} bg="#fff" s="0 1px 2px 0 rgba(0, 0, 0, 0.12)">
         <Striper>
          {report.map(item => (
            <BlockContent key={Math.random()}>
            <Card r={10}>
              {/* @ts-ignore */}
              <FlexItem>{`VSC: ${item.vcs}, repo: ${item.repo.replaceAll('%2F', '/')}`}</FlexItem>
            </Card>
            </BlockContent>
          ))}
         </Striper>
    </Card>
  )
}
