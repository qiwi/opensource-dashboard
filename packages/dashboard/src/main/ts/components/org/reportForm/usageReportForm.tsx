import React from 'react'
import { Box } from '@qiwi/pijma-core'
import { RadioField, TextField } from '@qiwi/pijma-desktop'
import { TReportProps } from './common'

export const UsageReportForm = ({ data, setData }: TReportProps) => (
  <div>
    <Box mb={'24px'} width={64}>
      <TextField
        title={'packageNamePattern'}
        type="text"
        value={data.packageNamePattern}
        onChange={(data) => setData({ type: 'packageNamePattern', data })}
      />
    </Box>
    <Box mb={'24px'} width={64}>
      <TextField
        title={'versionRange'}
        type="text"
        value={data.versionRange || ''}
        onChange={(data) => setData({ type: 'versionRange', data })}
      />
    </Box>
    <Box mb={'24px'}>
      <RadioField
        title="depType"
        options={[
          {
            label: 'optional',
            value: 'optional',
          },
          {
            label: 'peer',
            value: 'peer',
          },
          {
            label: 'default',
            value: 'default',
          },
          {
            label: 'dev',
            value: 'dev',
          },
          {
            label: 'all',
            value: 'all',
          },
        ]}
        value={data.depType}
        onChange={(data) => setData({ type: 'depType', data })}
      />
    </Box>
    <Box mb={'24px'}>
      <RadioField
        title="source"
        options={[
          {
            label: 'auto',
            value: 'auto',
          },
          {
            label: 'lock',
            value: 'lock',
          },
          {
            label: 'package',
            value: 'package',
          },
        ]}
        value={data.source}
        onChange={(data) => setData({ type: 'source', data })}
      />
    </Box>
  </div>
)
