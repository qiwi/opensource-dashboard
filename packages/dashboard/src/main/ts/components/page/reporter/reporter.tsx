import { Block, Card, Flex, Box } from '@qiwi/pijma-core'
import { Button, HeaderMenu } from '@qiwi/pijma-desktop'
import React, { useReducer } from 'react'
import { reportService } from '../../../service'
import { useAsync } from '../../../hooks/useAsyncExecutor'

import {
  TreeReportForm,
  VersionsReportForm,
  UsageReportForm,
  TReportProps,
} from '../../org/reportForm'
import { defaultTypes, treeDefault } from './constants'

function downloadObjectAsJson(exportObj: any, exportName: string) {
  const dataStr =
    'data:text/json;charset=utf-8,' +
    encodeURIComponent(JSON.stringify(exportObj))
  const downloadAnchorNode = document.createElement('a')
  downloadAnchorNode.setAttribute('href', dataStr)
  downloadAnchorNode.setAttribute('download', exportName + '.json')
  document.body.appendChild(downloadAnchorNode) // required for firefox
  downloadAnchorNode.click()
  downloadAnchorNode.remove()
}

const reducer = (
  state: Record<string, string>,
  action: Record<string, string>,
) => {
  if (action.type === 'type') {
    return defaultTypes[action.data as 'usage' | 'versions' | 'tree']
  }

  return { ...state, [action.type]: action.data }
}

const getReport = async ({
  type,
  depType,
  packageNamePattern,
  source,
  versionRange,
}: any) => {
  versionRange = versionRange || undefined
  const result = await reportService.getReport(type, {
    depType,
    packageNamePattern,
    source,
    versionRange,
  })

  downloadObjectAsJson(result, `${type}Report`)
}

export const Reporter = () => {
  const [data, setData] = useReducer(reducer, treeDefault)
  const [execute, status] = useAsync(getReport)
  return (
    <Flex mt={6}>
      <Block css={{ width: '100%', padding: '48px' }}>
        <Card width={1} height={14} mb={'32px'}>
          <HeaderMenu>
            {[
              {
                title: 'tree',
                active: data.type === 'tree',
                onClick: () => setData({ type: 'type', data: 'tree' }),
              },
              {
                title: 'usage',
                active: data.type === 'usage',
                onClick: () => setData({ type: 'type', data: 'usage' }),
              },
              {
                title: 'versions',
                active: data.type === 'versions',
                onClick: () => setData({ type: 'type', data: 'versions' }),
              },
            ]}
          </HeaderMenu>
        </Card>
        {
          {
            tree: (
              <TreeReportForm
                data={data as TReportProps['data']}
                setData={setData}
              />
            ),
            usage: (
              <UsageReportForm
                data={data as TReportProps['data']}
                setData={setData}
              />
            ),
            versions: (
              <VersionsReportForm
                data={data as TReportProps['data']}
                setData={setData}
              />
            ),
          }[data.type as 'usage' | 'versions' | 'tree']
        }
        <Box width={64}>
          <Button
            loading={status === 'pending'}
            disabled={status === 'pending'}
            type="button"
            kind="brand"
            size="normal"
            text="Получить отчет"
            onClick={() => execute(data)}
          />
        </Box>
      </Block>
    </Flex>
  )
}
