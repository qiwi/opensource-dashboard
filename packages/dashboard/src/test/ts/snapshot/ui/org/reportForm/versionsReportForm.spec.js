import React from 'react'
import { PijmaContext } from '../../../../pijmaContext'
import { render } from '@testing-library/react'
import { VersionsReportForm } from '../../../../../../main/ts/components/org/reportForm/'
import { versionsDefault } from '../../../../../../main/ts/components/page/reporter'
import '@testing-library/jest-dom/extend-expect'

describe('VersionsReportForm', () => {
  it('renders correctly', () => {
    const { container } = render(
      <PijmaContext>
        <VersionsReportForm
          data={versionsDefault}
          setData={() => {
            /* noop */
          }}
        />
      </PijmaContext>,
    )
    expect(container).toMatchSnapshot()
  })
})
