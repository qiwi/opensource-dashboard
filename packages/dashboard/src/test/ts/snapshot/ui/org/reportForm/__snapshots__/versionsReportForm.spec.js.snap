// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionsReportForm renders correctly 1`] = `
<div>
  <div>
    <div
      class="css-10ic8ni"
    >
      <div
        class="css-1haykwu"
      >
        <div
          class="css-1hy2az2"
        >
          <div
            class="css-dlnwt1"
          >
            <label
              class="css-195rf93"
            >
              packageNamePattern
            </label>
          </div>
        </div>
        <div
          class="css-1bx5ylf"
        >
          <div
            class="css-70qvj9"
          >
            <input
              autocomplete="off"
              class="css-amr1ku"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="css-1abbrc"
        />
      </div>
    </div>
    <div
      class="css-10ic8ni"
    >
      <div
        class="css-1haykwu"
      >
        <div
          class="css-1hy2az2"
        >
          <div
            class="css-dlnwt1"
          >
            <label
              class="css-195rf93"
            >
              versionRange
            </label>
          </div>
        </div>
        <div
          class="css-1bx5ylf"
        >
          <div
            class="css-70qvj9"
          >
            <input
              autocomplete="off"
              class="css-amr1ku"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="css-1abbrc"
        />
      </div>
    </div>
    <div
      class="css-adxqe4"
    >
      <div
        class="css-13o7eu2"
      >
        <div
          class="css-2stwun"
        >
          <div
            class="css-15rwwo"
          >
            <label
              class="css-1yifgsp"
            >
              depType
            </label>
          </div>
        </div>
        <div
          class="css-tprhxn"
        >
          <div
            class="css-13o7eu2"
            tabindex="0"
          >
            <div
              class="css-2stwun"
            >
              <div
                class="css-z0xd9q"
              >
                <div
                  class="css-196huyw"
                >
                  <div
                    class="css-1mss6i9"
                  >
                    <div
                      class="css-110z6b4"
                    >
                      <svg
                        class="css-nmzr3v"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          class="css-qz95lw"
                          fill="#e6e6e6"
                          height="18"
                          rx="9"
                          ry="9"
                          width="18"
                          x="3"
                          y="3"
                        />
                        <path
                          class="css-qz95lw"
                          d="M12 3a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 2a7 7 0 1 0 0 14 7 7 0 0 0 0-14z"
                          fill="transparent"
                        />
                        <rect
                          class="css-qz95lw"
                          fill="transparent"
                          height="8"
                          rx="4"
                          ry="4"
                          width="8"
                          x="8"
                          y="8"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-59tnyu"
                    >
                      optional
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-18ifh88"
            >
              <div
                class="css-z0xd9q"
              >
                <div
                  class="css-196huyw"
                >
                  <div
                    class="css-1mss6i9"
                  >
                    <div
                      class="css-110z6b4"
                    >
                      <svg
                        class="css-nmzr3v"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          class="css-qz95lw"
                          fill="#e6e6e6"
                          height="18"
                          rx="9"
                          ry="9"
                          width="18"
                          x="3"
                          y="3"
                        />
                        <path
                          class="css-qz95lw"
                          d="M12 3a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 2a7 7 0 1 0 0 14 7 7 0 0 0 0-14z"
                          fill="transparent"
                        />
                        <rect
                          class="css-qz95lw"
                          fill="transparent"
                          height="8"
                          rx="4"
                          ry="4"
                          width="8"
                          x="8"
                          y="8"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-59tnyu"
                    >
                      peer
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-18ifh88"
            >
              <div
                class="css-z0xd9q"
              >
                <div
                  class="css-196huyw"
                >
                  <div
                    class="css-1mss6i9"
                  >
                    <div
                      class="css-110z6b4"
                    >
                      <svg
                        class="css-nmzr3v"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          class="css-qz95lw"
                          fill="#e6e6e6"
                          height="18"
                          rx="9"
                          ry="9"
                          width="18"
                          x="3"
                          y="3"
                        />
                        <path
                          class="css-qz95lw"
                          d="M12 3a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 2a7 7 0 1 0 0 14 7 7 0 0 0 0-14z"
                          fill="transparent"
                        />
                        <rect
                          class="css-qz95lw"
                          fill="#000"
                          height="8"
                          rx="4"
                          ry="4"
                          width="8"
                          x="8"
                          y="8"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-59tnyu"
                    >
                      default
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-18ifh88"
            >
              <div
                class="css-z0xd9q"
              >
                <div
                  class="css-196huyw"
                >
                  <div
                    class="css-1mss6i9"
                  >
                    <div
                      class="css-110z6b4"
                    >
                      <svg
                        class="css-nmzr3v"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          class="css-qz95lw"
                          fill="#e6e6e6"
                          height="18"
                          rx="9"
                          ry="9"
                          width="18"
                          x="3"
                          y="3"
                        />
                        <path
                          class="css-qz95lw"
                          d="M12 3a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 2a7 7 0 1 0 0 14 7 7 0 0 0 0-14z"
                          fill="transparent"
                        />
                        <rect
                          class="css-qz95lw"
                          fill="transparent"
                          height="8"
                          rx="4"
                          ry="4"
                          width="8"
                          x="8"
                          y="8"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-59tnyu"
                    >
                      dev
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-tprhxn"
            >
              <div
                class="css-z0xd9q"
              >
                <div
                  class="css-196huyw"
                >
                  <div
                    class="css-1mss6i9"
                  >
                    <div
                      class="css-110z6b4"
                    >
                      <svg
                        class="css-nmzr3v"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          class="css-qz95lw"
                          fill="#e6e6e6"
                          height="18"
                          rx="9"
                          ry="9"
                          width="18"
                          x="3"
                          y="3"
                        />
                        <path
                          class="css-qz95lw"
                          d="M12 3a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 2a7 7 0 1 0 0 14 7 7 0 0 0 0-14z"
                          fill="transparent"
                        />
                        <rect
                          class="css-qz95lw"
                          fill="transparent"
                          height="8"
                          rx="4"
                          ry="4"
                          width="8"
                          x="8"
                          y="8"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-59tnyu"
                    >
                      all
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-adxqe4"
    >
      <div
        class="css-13o7eu2"
      >
        <div
          class="css-2stwun"
        >
          <div
            class="css-15rwwo"
          >
            <label
              class="css-1yifgsp"
            >
              source
            </label>
          </div>
        </div>
        <div
          class="css-tprhxn"
        >
          <div
            class="css-13o7eu2"
            tabindex="0"
          >
            <div
              class="css-2stwun"
            >
              <div
                class="css-z0xd9q"
              >
                <div
                  class="css-196huyw"
                >
                  <div
                    class="css-1mss6i9"
                  >
                    <div
                      class="css-110z6b4"
                    >
                      <svg
                        class="css-nmzr3v"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          class="css-qz95lw"
                          fill="#e6e6e6"
                          height="18"
                          rx="9"
                          ry="9"
                          width="18"
                          x="3"
                          y="3"
                        />
                        <path
                          class="css-qz95lw"
                          d="M12 3a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 2a7 7 0 1 0 0 14 7 7 0 0 0 0-14z"
                          fill="transparent"
                        />
                        <rect
                          class="css-qz95lw"
                          fill="#000"
                          height="8"
                          rx="4"
                          ry="4"
                          width="8"
                          x="8"
                          y="8"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-59tnyu"
                    >
                      auto
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-18ifh88"
            >
              <div
                class="css-z0xd9q"
              >
                <div
                  class="css-196huyw"
                >
                  <div
                    class="css-1mss6i9"
                  >
                    <div
                      class="css-110z6b4"
                    >
                      <svg
                        class="css-nmzr3v"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          class="css-qz95lw"
                          fill="#e6e6e6"
                          height="18"
                          rx="9"
                          ry="9"
                          width="18"
                          x="3"
                          y="3"
                        />
                        <path
                          class="css-qz95lw"
                          d="M12 3a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 2a7 7 0 1 0 0 14 7 7 0 0 0 0-14z"
                          fill="transparent"
                        />
                        <rect
                          class="css-qz95lw"
                          fill="transparent"
                          height="8"
                          rx="4"
                          ry="4"
                          width="8"
                          x="8"
                          y="8"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-59tnyu"
                    >
                      lock
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-tprhxn"
            >
              <div
                class="css-z0xd9q"
              >
                <div
                  class="css-196huyw"
                >
                  <div
                    class="css-1mss6i9"
                  >
                    <div
                      class="css-110z6b4"
                    >
                      <svg
                        class="css-nmzr3v"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          class="css-qz95lw"
                          fill="#e6e6e6"
                          height="18"
                          rx="9"
                          ry="9"
                          width="18"
                          x="3"
                          y="3"
                        />
                        <path
                          class="css-qz95lw"
                          d="M12 3a9 9 0 1 1 0 18 9 9 0 0 1 0-18zm0 2a7 7 0 1 0 0 14 7 7 0 0 0 0-14z"
                          fill="transparent"
                        />
                        <rect
                          class="css-qz95lw"
                          fill="transparent"
                          height="8"
                          rx="4"
                          ry="4"
                          width="8"
                          x="8"
                          y="8"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-59tnyu"
                    >
                      package
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
